src/FaultCheck.hpp

```diff
@@ -0,0 +1,15 @@
+#ifndef FAULT_CHECK_HPP_
+#define FAULT_CHECK_HPP_
+
+#include "sfr.hpp"
+
+class FaultCheck
+{
+public:
+    static bool check_is_fault(fault_check_type fault_type);
```

Comment:
You should not need a custom type. The majority of sensor values are numeric so you should be able to cast everything to a float.